src/main/scala/mimir/adaptive/CheckHeader.scala

```diff
@@ -0,0 +1,341 @@
+package mimir.adaptive
+
+import java.io._
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import scala.collection.JavaConverters._
+
+import mimir.Database
+import mimir.algebra._
+import mimir.lenses._
+import mimir.models._
+import mimir.statistics.FuncDep
+
+object CheckHeader
+  extends Multilens
+  with LazyLogging
+{
+  def checkheader(db: Database, config: MultilensConfig): Boolean = {
+    val col = config.query.expressions(0);
+    val tablename = col.toString.split("_")(0);
```

Comment:
1. Please do not assume that the input query is a single table.
2. Please do not use string manipulations to extract content from structured data.  General rule: If you are ever using toString for a purpose other than producing output for human consumption, you're probably doing it wrong.  This is what the [scala match](http://tutorials.jenkov.com/scala/match.html) operator is for.
3. db.query() accepts operators.  Please manipulate Operators directly (e.g., several common manipulations have convenience functions defined in https://github.com/UBOdin/mimir/blob/master/src/main/scala/mimir/algebra/OperatorConstructors.scala ).